packages/Video/Video.jsx

```diff
@@ -311,6 +316,44 @@ class Video extends React.Component {
     this.clearInactivityTimer()
   }
 
+  updateAnalyticsData = () => {
+    const analyticsObject = {}
+    analyticsObject.action = this.state.videoIsPlaying ? 'pause' : 'play'
+    this.props.analyticsTracking(analyticsObject)
+  }
+
+  // required for analytics
+  calculatePercentageWatched = () => {
+    let percentValue = (this.state.videoCurrentTime / this.state.videoLength) * 100
+    percentValue = Math.round(percentValue)
+    const previousValue = this.state.percentPlayed
+    this.percentageBucket(percentValue)
+    if (previousValue !== this.state.percentPlayed) {
+      const analyticsObject = {}
+      analyticsObject.action = this.state.percentPlayed
+      this.props.analyticsTracking(analyticsObject)
+    }
+  }
+
+  percentageBucket = percentValue => {
+    if (percentValue < 25) {
+      return this.setState({ percentPlayed: 'watched 0%' })
+    }
+    if (percentValue < 50) {
+      return this.setState({ percentPlayed: 'watched 25%' })
+    }
+    if (percentValue < 75) {
+      return this.setState({ percentPlayed: 'watched 50%' })
+    }
+    if (percentValue < 100) {
+      return this.setState({ percentPlayed: 'watched 75%' })
+    }
+    if (percentValue === 100) {
+      return this.setState({ percentPlayed: 'watched 100%' })
+    }
+    return false
+  }
+
```

Comment:
Would it be better to return `elapsedTime` and `totalTime` instead of `percentPlayed`. This way outcome teams can implement their own interpretation of the metrics.

Reply:
Hi Jordan, I can change percentPlayed to elapsedTime no problem. There's no need for a totalTime, because "videoLength" already exists.